content/concepts/did-ddo.md

```diff
@@ -35,97 +70,178 @@ In Ocean, a DID is a string that looks like:
 
-Each service is distinguished by the `DDO.service.type` attribute.
+### Algorithm attributes
 
-Each service has an `attributes` section holding the information related to the service. That section _must_ have a `main` sub-section, holding all the mandatory information that a service has to provide.
+An asset of type `algorithm` has the following additional attributes under `algorithm` in metadata object:
 
-A part of the `attributes.main` sub-section, other optional sub-sections like `attributes.extra` can be added. These depend on the service type.
+| Attribute           |   Type                |   Required  | Description                                         |
+| ------------------- | ----------------------| ----------- |--------------------------------------------------- |
+| **`language`**      | `string`              | no          | Language used to implement the software |
+| **`version`**       | `string`              | no          | Version of the software. |
+| **`container`**     | `Container Object`    | yes         | Object describing the Docker container image.(see below) |
 
-Each service has a `timeout` (in seconds) section describing how long the sevice can be used after consumption is initiated. A timeout of 0 represents no time limit.
+The `container` object has the following attributes:
 
-The `cost` attribute is obsolete, as of Ocean V3. As of V3, to consume an asset, one sends exactly 1.0 datatokens of the asset, so a `cost` is not needed.
+| Attribute           |   Type   | Required  | Description                                         |
+| ------------------- | -------- | --------- | --------------------------------------------------- |
+| **`entrypoint`**    | `string` | yes       | The command to execute, or script to run inside the Docker image. |
+| **`image`**         | `string` | yes       | Name of the Docker image. |
+| **`tag`**           | `string` | yes       | Tag of the Docker image. |
+| **`checksum`**      | `string` | yes       | Checksum of the Docker image. |
 
-## DDO Service Example
 
-Here is an example DDO service:
+
+
+
+## Services
+
+| Attribute           | Type                  | Required | Description                                                                                                                                                                                       |
+| ------------------- | --------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **`type`**            | Text          | **Yes**       | Type of service (access, compute, wss, etc |
+| **`name`**            | Text          | No       | Service friendly name |
+| **`description`**     | Text          | No       | Service description |
+| **`datatokenAddress`**     | Text          | Yes       | Datatoken address |
```

Comment:
getting name & symbol is actually not fun on frontend so would be handy keeping this as an object as we had. Also during publish we already have the name & symbol too so we can simply add it so would be no need for Aquarius to fetch it itself.

```
datatoken: {
  address: string
  name: string
  symbol: string
}
```

Reply:
I'm still keeping the name and symbol as it previously was. But I don't have number of decimals now, since that's on the template now.

Reply:
but if you implement then would be great to also add it to the spec. And based on spec it seems weird keeping `dataTokenOptions` on the root as data tokens are now attached to each service (so could be multiple data tokens for one dataset, what then?). 

Also thought about this:

> Also during publish we already have the name & symbol too so we can simply add it so would be no need for Aquarius to fetch it itself.

Actually the source of truth for that name & symbol should be the chain. Don't know if that is a weird attack but user could create the datatoken with another name & symbol than what we have sent to Aquarius. So where does Aquarius get it from now?